### PR TITLE
Many typo fixes.

### PR DESCRIPTION
**Pull request by cdluminate**
Debian's lintian package checker just complains a lot about typos.

Credit: https://github.com/codespell-project/codespell/

### CODE REVIEW

File: stdlib/Dates/docs/src/index.md [hunk 1]
```diff
@@ -427,7 +427,7 @@ julia> collect(dr)
 
 ## Adjuster Functions
 
-As convenient as date-period arithmetics are, often the kinds of calculations needed on dates
+As convenient as date-period arithmetic are, often the kinds of calculations needed on dates
```
Comment (rfourquet) [inline stdlib/Dates/docs/src/index.md]: "arithmetic is" ?

File: stdlib/Pkg/src/GraphType.jl [hunk 2]
```diff
@@ -15,7 +15,7 @@ export Graph, ResolveLog, add_reqs!, add_fixed!, simplify_graph!, simplify_graph
 # a message for the log.
 #
 # Specialized functions called `log_event_[...]!` are used to store the
-# various events. The events are also recorded in orded in a shared
+# various events. The events are also recorded in ordered in a shared
```
Comment (rfourquet) [inline stdlib/Pkg/src/GraphType.jl]: I guess here should be "order"

**Review by rfourquet — APPROVED** [review]
Amazing, thank you! Besides the two comments I left, loooks good.

File: stdlib/Dates/docs/src/index.md (hunk 1 above)
Comment (CDLuminate) [reply stdlib/Dates/docs/src/index.md]: Reverted in https://github.com/JuliaLang/julia/pull/27976/commits/c551994938c7ffaebf79113dca685be26c190951

File: stdlib/Pkg/src/GraphType.jl (hunk 2 above)
Comment (CDLuminate) [reply stdlib/Pkg/src/GraphType.jl]: Yes, fixed.

**Comment by KristofferC** [comment]
Should maybe not change things in `deps/patches`?

**Comment by CDLuminate** [comment]
Updated as requested. Not touching any file under `deps/patches` . Is it acceptable now?

**Comment by ViralBShah** [comment]
Looks great. We can merge as soon as CI passes.

**Comment by ViralBShah** [comment]
Failures appear spurious and unrelated in both cases.

**Comment by ViralBShah** [comment]
That's a nifty tool!

**Pull request merged**